builders/integrations/oracles/dia.md

```diff
@@ -0,0 +1,139 @@
+
+By collecting billions of raw trades directly from over **90 sources, including CEXs, DEXs, and NFT marketplaces**, DIA enables full transparency, customization, and control throughout the entire value stack. DIA's data and oracle suite comprise **price feeds for 20,000+ assets** including cryptocurrencies, NFT collections, and liquid-staked tokens, as well as random number generation and other data feed types.
+
+→ [Request a Custom Oracle | DIA Documentation](https://docs.diadata.org/introduction/intro-to-dia-oracles/request-an-oracle)
+
+# 🪙 Token Price Feeds
+
+DIA token price feeds provide smart contracts with real-time price information for [3,000+ cryptocurrencies](https://diadata.org/app/price), sourced transparently from [90+ trusted, high-volume DEXs and CEXs](https://diadata.org/app/source/defi).
+
+## How to access DIA oracles?
+
+Here is an example of how to access a price value on DIA oracles:
+
+1. Access your custom oracle smart contract on Moonbeam.
+2. Call `getValue(pair_name)` with `pair_name` being the full pair name such as `BTC/USD`. You can use the "Read" section on the explorer to execute this call.
+3. The response of the call contains two values:
+- The current asset price in USD with a fix-comma notation of 8 decimals.
+- The UNIX timestamp of the last oracle update.
+
+You can find DIA's oracle integration samples in Solidity and Vyper languages by visiting:
+
+→ [Access the Oracle | DIA Documentation](https://docs.diadata.org/products/token-price-feeds/access-the-oracle)
+
+## Moonbeam demo price oracles
+
+DIA has deployed the following demo oracles for the Moonbeam community. It provides a limited selection of cryptocurrency price feeds with predefined configuration settings.
+
+> ⚠️ NOTE: DIA demo oracles are not intended for use in production environments. Developers can request a dedicated, production-ready oracle with custom price feeds and configuration settings. Start the request process: [Request a Custom Oracle | DIA Documentation](https://docs.diadata.org/introduction/intro-to-dia-oracles/request-an-oracle)
+
+### Demo Oracle Smart Contracts
+
+| Network        | Contract address      
+| -------------- | -------------------------------------------------------------------------------------------------------------------------------------- |
+| Moonbeam Alpha  | [`0xe23d8713aa3a0a2c102af772d2467064821b8d46`](https://moonbase.moonscan.io/address/0xe23d8713aa3a0a2c102af772d2467064821b8d46)        |
+| Moonriver  | [`0x11f74b94afb5968119c98ea277a2b73208bb39ab`](https://moonriver.moonscan.io/address/0x11f74b94afb5968119c98ea277a2b73208bb39ab)        |
```

Comment:
Is there one for Moonbeam?

Reply:
Unfortunately, we currently don't have a demo oracle for Moonbeam, as these are just for testing. DIA deploys custom oracles for every dApp and use case, to create the best data solution. That's why we believe it's sufficient to have a demo oracle on Moonbase Alpha. But that is also something we could set up if required!